pkg/logs/config/source.go

```diff
@@ -9,16 +9,22 @@ import (
 	"sync"
 )
 
+// Types of log formats that are used to determine which parser is going to be used
+const (
+	ContainerdFormat = "containerd"
+)
+
 // LogSource holds a reference to an integration name and a log configuration, and allows to track errors and
 // successful operations on it. Both name and configuration are static for now and determined at creation time.
 // Changing the status is designed to be thread safe.
 type LogSource struct {
-	Name     string
-	Config   *LogsConfig
-	Status   *LogStatus
-	inputs   map[string]bool
-	lock     *sync.Mutex
-	Messages *Messages
+	Name         string
+	Config       *LogsConfig
+	Status       *LogStatus
+	inputs       map[string]bool
+	lock         *sync.Mutex
+	Messages     *Messages
+	parserFormat string
```

Comment:
I think we can avoid this altogether by moving the check to the container id in the tailer.